R/file_analysis.R

```diff
@@ -0,0 +1,59 @@
+source("R/connect.R")
+source("R/utils.R")
+
+#' Sends a ["file-analysis"](https://github.com/flowr-analysis/flowr/wiki/Interface#the-analysis-request) request
+#'
+#' Note that either `filepath` or `content` must be provided, but never both.
+#'
+#' @param con The connection to the server
+#' @param filepath The path to the file (must be visible to the server)
+#' @param content The R code to be analyzed
+#' @param cfg Weather to include the control flow graph in the response
+#' @param id The id of the request
+#' @param filetoken The filetoken of the file to be analyzed
+#' @return A list containing the id, filetoken and the response
+#'
+#' @seealso [connect()]
+#'
+#' @export
+request_file_analysis <- function(con,
+                                  filepath = NULL,
+                                  content = NULL,
+                                  cfg = FALSE,
+                                  id = get_new_id(),
+                                  filetoken = get_filetoken(filepath, content)) {
+  if (is.null(filepath) && is.null(content)) {
```

Comment:
Type checking goes here as well (i.e., no null, but type checks). Additionally, we may want to use `stopifnot`?

Reply:
Resolved in 569eb17